### PR TITLE
Block named and looks like I was added. TestNG upgraded.

### PR DESCRIPTION
**Pull request by KrzysztofDolanski**

### CODE REVIEW

File: src/main/java/com/epam/prejap/teatrees/block/BlockFeed.java [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
     private final Random rnd = new Random();
     private final List<Supplier<Block>> blocks = List.of(
-        OBlock::new
+        OBlock::new, IBlock::new
```
Comment (pkierat-epam) [inline src/main/java/com/epam/prejap/teatrees/block/BlockFeed.java:11]: This change is nowhere tested.

File: src/test/java/com/epam/prejap/teatrees/block/IBlockTest.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+public class IBlockTest {
+
+
+    @Test
+    public void imageShouldBeTheSameAsGivenGeometricalFigure(){
+
+        // Given
+        Block iBlock = new IBlock();
+        // When
+        int i0j0 = 0;
```
Comment (pkierat-epam) [inline src/test/java/com/epam/prejap/teatrees/block/IBlockTest.java:18]: Couldn't this be written in a more compact way, say, with some loops?

**Review by pkierat-epam — CHANGES_REQUESTED** [review]

File: src/main/java/com/epam/prejap/teatrees/block/BlockFeed.java (hunk 1 above)
Comment (Nimrud) [inline src/main/java/com/epam/prejap/teatrees/block/BlockFeed.java:11]: new line please

File: src/test/java/com/epam/prejap/teatrees/block/IBlockTest.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
+package com.epam.prejap.teatrees.block;
+
+import org.testng.annotations.Test;
+import org.testng.asserts.SoftAssert;
+
+/**
```
Comment (Nimrud) [inline src/test/java/com/epam/prejap/teatrees/block/IBlockTest.java:6]: No Javadoc.

File: src/test/java/com/epam/prejap/teatrees/block/IBlockTest.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+ * @author Marta Krawczykowska, Krzysztof Dolanski
+ */
+public class IBlockTest {
```
Comment (Nimrud) [inline src/test/java/com/epam/prejap/teatrees/block/IBlockTest.java:9]: Why only 1 unit test, why no integration tests?

**Review by Nimrud — CHANGES_REQUESTED** [review]

File: src/test/java/com/epam/prejap/teatrees/IBlockTest.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+/**
+ * @author Marta Krawczykowska, Krzysztof Dolanski
+ */
+public class IBlockTest {
+
+
+    @Test
+    public void imageShouldBeTheSameAsGivenGeometricalFigure(){
+
```
Comment (emmaline261b) [inline src/test/java/com/epam/prejap/teatrees/IBlockTest.java]: IMO could be clearer if you used an array or something to show the shape of the block?

File: src/test/java/com/epam/prejap/teatrees/block/IBlockTest.java (hunk 2 above)
Comment (KrzysztofDolanski) [reply src/test/java/com/epam/prejap/teatrees/block/IBlockTest.java]: Done by usage of @DataProvider

File: src/test/java/com/epam/prejap/teatrees/IBlockTest.java (hunk 5 above)
Comment (KrzysztofDolanski) [reply src/test/java/com/epam/prejap/teatrees/IBlockTest.java]: Done by usage of @DataProvider. Thanks for a hint.

File: src/main/java/com/epam/prejap/teatrees/block/BlockFeed.java (hunk 1 above)
Comment (KrzysztofDolanski) [reply src/main/java/com/epam/prejap/teatrees/block/BlockFeed.java:11]: Done.

File: src/test/java/com/epam/prejap/teatrees/block/IBlockTest.java (hunk 4 above)
Comment (KrzysztofDolanski) [reply src/test/java/com/epam/prejap/teatrees/block/IBlockTest.java:11]: Done.

**Pull request merged**